tests/tablespaces/tablespace_utils.py

```diff
@@ -0,0 +1,113 @@
+import time
+import datetime as dt
+import psycopg2
+
+
+class PGNodeTS:
+    def __init__(
+        self,
+        port,
+        service_name,
+        monitor_node=None,
+    ):
+        self.port = port
+        self.service_name = service_name
+        self.monitor = monitor_node
+        if monitor_node is None:
+            self.monitor = self
+
+    def run_sql_query(self, query, *args):
+        """
+        Runs the given sql query with the given arguments in this postgres node
+        and returns the results. Returns None if there are no results to fetch.
+        """
+        result = None
+        conn = psycopg2.connect(self.connection_string())
+        conn.autocommit = True
+
+        with conn:
+            with conn.cursor() as cur:
+                cur.execute(query, args)
+                try:
+                    result = cur.fetchall()
+                except psycopg2.ProgrammingError:
+                    pass
+        # leaving contexts closes the cursor, however
+        # leaving contexts doesn't close the connection
+        conn.close()
+
+        return result
+
+    def connection_string(self):
+        if self.service_name == "monitor":
+            return (
+                "postgresql://autoctl_node@"
+                + self.service_name
+                + ":"
+                + self.port
+                + "/pg_auto_failover"
+            )
+        else:
+            return (
+                "postgresql://docker@"
+                + self.service_name
+                + ":"
+                + self.port
+                + "/postgres"
+            )
```

Comment:
Same worries again duplication of code and maintenance compared to enhancing the existing class or inheriting from it.